presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -334,12 +336,20 @@ private boolean addIncludedStreams(Map<Integer, ColumnEncoding> columnEncodings,
             SharedBuffer sharedDecompressionBuffer)
             throws IOException
     {
-        //
-        // Note: this code does not use the Java 8 stream APIs to avoid any extra object allocation
-        //
+        // get index streams from the stripe meta cache
+        Map<StreamId, OrcDataSourceInput> cachedInputs = getCachedIndexStreams(stripeId, diskRanges);
```

Comment:
This is where we query the stripe metadata cache first. We would then proceed to read the rest of uncached index data directly from the source.